app/controllers/printer_controller.rb

```diff
@@ -0,0 +1,34 @@
+class PrinterController < ApplicationController
+  before_action :authenticate_user, :get_booking
+
+  def print
+    render pdf: "print", layout: "pdf.html.erb"
+  end
+
+  def download
+    html = render_to_string("print", layout: "pdf.html.erb")
+    pdf = WickedPdf.new.pdf_from_string(html)
+    send_data(pdf, filename: "#{@booking.event.title.downcase}.pdf")
+  end
+
+
+  private
+
+  def get_booking
+    @booking = current_user.bookings.find_by(id: params[:booking_id])
+    if @booking.nil?
+      flash[:notice] = 'Booking not found'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.